crates/bitwarden-vault/src/cipher/migrations/versions/v2_to_v3.rs

```diff
@@ -0,0 +1,118 @@
+use base64::{prelude::BASE64_STANDARD, Engine};
+use bitwarden_core::key_management::{KeyIds, SymmetricKeyId};
+use bitwarden_crypto::{Decryptable, EncString, KeyStoreContext, PrimitiveEncryptable};
+
+use crate::{migrations::registry::Migration, CipherError};
+
+pub struct V2ToV3Migration;
+
+impl Migration for V2ToV3Migration {
+    fn source_version(&self) -> u32 {
+        2
+    }
+
+    fn target_version(&self) -> u32 {
+        3
+    }
+
+    fn migrate(
+        &self,
+        cipher_data: &mut serde_json::Value,
+        ctx: Option<&mut KeyStoreContext<KeyIds>>,
+        cipher_key: Option<SymmetricKeyId>,
+    ) -> Result<(), CipherError> {
+        let ctx =
+            ctx.ok_or_else(|| CipherError::MigrationFailed("Crypto context required".to_string()))?;
+
+        let ciphers_key = cipher_key
+            .ok_or_else(|| CipherError::MigrationFailed("Cipher key required".to_string()))?;
+
+        if let Some(fido2_credentials) = cipher_data
+            .get_mut("fido2Credentials")
+            .and_then(|v| v.as_array_mut())
+        {
+            for fido2_credential in fido2_credentials {
+                if let Some(credential_id_str) = fido2_credential
+                    .get("credentialId")
+                    .and_then(|v| v.as_str())
+                {
+                    let enc_string: EncString = credential_id_str.parse()?;
+                    let dec_credential_id: String = enc_string.decrypt(ctx, ciphers_key)?;
+                    let b64_credential_id = BASE64_STANDARD.encode(&dec_credential_id);
+                    let enc_credential_id: EncString =
+                        b64_credential_id.encrypt(ctx, ciphers_key)?;
```

Comment:
issue: I think this migration does a good job as an example of how one would modify a field, but the what it actually does (since this is a real example) is wrong. Probably don't need to fix it in the PoC, but this should not be merged to production as-is.

**What it does:**
It takes the UUID as a string directly and encodes the Unicode bytes as B64 i.e. `UUID -> B64`.
E.g. the UUID standard format `0000 ...` will be converted to the unicode bytes `0x30 0x30 0x30 0x30 ...` and saved as B64

What it should do:
Take the UUID standard format and convert it into raw format e.g. `0000 ...` means `0x00 0x00` and then save that as B64